src/Perl6/Metamodel/CurriedRoleHOW.nqp

```diff
@@ -60,7 +69,56 @@ class Perl6::Metamodel::CurriedRoleHOW
         my $meta := self.new(:curried_role($curried_role), :pos_args(@pos_args),
             :named_args(%named_args), :name($name));
         my $type := nqp::settypehll(nqp::newtype($meta, 'Uninstantiable'), 'perl6');
-        nqp::settypecheckmode($type, 2)
+        nqp::settypecheckmode($type, 2);
+    }
+
+    method parameterize_roles($obj) {
+        my @pos_args;
+        nqp::push(@pos_args, $obj);
+        for @!pos_args {
+            nqp::push(@pos_args, $_);
+        }
+        my $candidate := $!curried_role.HOW.select_candidate($!curried_role, @pos_args, %!named_args);
+        my $type_env;
+        my $error;
+        try {
+            my $body_block := nqp::decont($candidate.HOW.body_block($candidate));
+            my @result := $candidate.HOW.body_block($candidate)(|@pos_args, |%!named_args);
+            $type_env := @result[1];
+        }
+        my @r := $candidate.HOW.roles($candidate, :!transitive);
```

Comment:
This may just be personal preference, but I find the next couple of lines a little confusing. Why not `for $candidate.HOW.roles($candidate, :!transitive) -> $role { ... }`?

Reply:
@MasterDuke17 thanks! These all are leftovers from debugging. I happened to be too tired when comitting, as it seems, to wipe out all debug-related stuff.